scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/cipher/CCMBlockCipher.java

```diff
@@ -307,7 +324,7 @@ protected int xorInt(int offset, int end, int number) {
 		 */
 
 		// instantiate the underlying block cipher
-		Cipher cipher = Cipher.getInstance(CIPHER_NAME);
+		Cipher cipher = threadLocalCipher.get();
```

Comment:
if `Cipher.getInstance(CIPHER_NAME)` failed next line will raise an NPE. Should we handle this in a better way ?

Reply:
I'm not sure, if coaps makes sense without AES. 
So this would only report a nicer error in those cases, where dtls could not be used at all.

Reply:
I agree, but I think at least a log would be valuable

Reply:
OK, I will  add that.

Reply:
Done.